### PR TITLE
Housekeeping

### PR DESCRIPTION
**Pull request by mattruggio**
This is a general cleanup PR:

* Bump supported Ruby to >= 2.5
* Update development dependencies
* Install Rake / new publication guidelines
* Remove Gemfile.lock
* Remove un-necessary file indirection
* Remove un-necessary root namespacing
* Install Code of Conduct

### CODE REVIEW

File: lib/stringento.rb [hunk 1]
```diff
@@ -7,4 +7,40 @@
 # LICENSE file in the root directory of this source tree.
 #
 
-require_relative 'stringento/stringento'
```
Comment (mattruggio) [inline lib/stringento.rb:10]: This was useless indirection.

File: lib/stringento/version.rb [hunk 2]
```diff
@@ -8,5 +8,5 @@
 #
 
 module Stringento
-  VERSION = '2.0.0'
```
Comment (mattruggio) [inline lib/stringento/version.rb:11]: I struggle a little with what to bump when updating Ruby, but the [semver guide suggests it is a minor bump](https://semver.org/spec/v2.0.0.html#what-should-i-do-if-i-update-my-own-dependencies-without-changing-the-public-api).

File: stringento.gemspec [hunk 3]
```diff
@@ -5,28 +5,36 @@ require './lib/stringento/version'
 Gem::Specification.new do |s|
   s.name        = 'stringento'
   s.version     = Stringento::VERSION
-  s.summary     = 'Configurable Data Shaper'
+  s.summary     = 'This library provides a pluggable string templating system.'
 
   s.description = <<-DESCRIPTION
-    Stringento is a configuration-based object graphing tool which can turn a flat, single dimensional dataset into a structure of deeply nested objects.
```
Comment (mattruggio) [inline stringento.gemspec:11]: I noticed this was just flat wrong.

File: .travis.yml [hunk 4]
```diff
@@ -4,10 +4,9 @@ env:
 language: ruby
 rvm:
   # Build on the latest stable of all supported Rubies (https://www.ruby-lang.org/en/downloads/):
-  - 2.3.8
-  - 2.4.5
-  - 2.5.3
-  - 2.6.0
+  - 2.5.8
+  - 2.6.6
+  - 2.7.1
```
Comment (ddewar) [inline .travis.yml:9]: Is there a reason we're including 2.7.1 here?

Comment (mattruggio) [reply .travis.yml:9]: [2.7 was released about 8 months ago, 2.7.1 about 5 months ago](https://www.ruby-lang.org/en/news/2019/12/25/ruby-2-7-0-released/), so this is just looking forward a little and ensuring it builds properly for the most recent version

**Pull request merged**